### PR TITLE
[No ticket] Create google project sam resource when creating a v2 workspace

### PR DESCRIPTION
**Pull request by andy7i**
This adds the missing google-project sam resource when creating a workspace (in the first commit)

The second commit renames `SamProjectRoles` to `SamBillingProjectRoles` for clarity.

Might be worth creating a unit test for this so that we don't accidentally miss this / mess it up in the future

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 1]
```diff
@@ -2105,6 +2105,12 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
...
+                  // After the workspace has been created, create the google-project resource in Sam with the workspace as the resource parent
+                  val googleProjectOwnerPolicy = SamGoogleProjectPolicyNames.owner -> SamPolicy(Set.empty, Set.empty, Set(SamGoogleProjectRoles.owner))
+                  val googleProjectNotebookUserPolicy = SamGoogleProjectPolicyNames.notebookUser -> SamPolicy(Set.empty, Set.empty, Set(SamGoogleProjectRoles.notebookUser))
+                  val googleProjectPolicies = Map(googleProjectOwnerPolicy, googleProjectNotebookUserPolicy)
+                  traceDBIOWithParent("createResourceFull (google project)", s1)(_ => DBIO.from(samDAO.createResourceFull(SamResourceTypeNames.googleProject, googleProjectId.value, googleProjectPolicies, workspaceRequest.authorizationDomain.getOrElse(Set.empty).map(_.membersGroupName.value), userInfo, Option(SamFullyQualifiedResourceId(workspaceId, SamResourceTypeNames.workspace.value)))))
+
```
Comment (andy7i) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: this is the meat of the PR. I'm open to moving it elsewhere (or maybe into an inner function) if folks have suggestions.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/model/SamModel.scala [hunk 2]
```diff
@@ -105,6 +110,11 @@ object SamBillingProjectPolicyNames {
   val canComputeUser = SamResourcePolicyName("can-compute-user")
 }
 
+object SamGoogleProjectPolicyNames {
+  val owner = SamResourcePolicyName("owner")
+  val notebookUser = SamResourcePolicyName("notebook-user")
+}
+
```
Comment (dvoet) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/model/SamModel.scala]: we should not be adding any policies to google project resources, all policies should be inherited from the parent

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 3]
```diff
@@ -2105,6 +2105,12 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
...
               traceDBIOWithParent("setupGoogleProject", s1)(_ => DBIO.from(setupGoogleProject(billingProject, billingAccount, s1, workspaceId))).flatMap { case (googleProjectId, googleProjectNumber) =>
                 traceDBIOWithParent("saveNewWorkspace", s1)(s2 => saveNewWorkspace(workspaceId, workspaceRequest, bucketName, projectOwnerPolicyEmail, googleProjectId, Option(googleProjectNumber), Option(billingAccount), dataAccess, s2).flatMap { case (savedWorkspace, policyMap) =>
+                  // After the workspace has been created, create the google-project resource in Sam with the workspace as the resource parent
+                  val googleProjectOwnerPolicy = SamGoogleProjectPolicyNames.owner -> SamPolicy(Set.empty, Set.empty, Set(SamGoogleProjectRoles.owner))
+                  val googleProjectNotebookUserPolicy = SamGoogleProjectPolicyNames.notebookUser -> SamPolicy(Set.empty, Set.empty, Set(SamGoogleProjectRoles.notebookUser))
+                  val googleProjectPolicies = Map(googleProjectOwnerPolicy, googleProjectNotebookUserPolicy)
```
Comment (dvoet) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: don't need these

**Review by dvoet — CHANGES_REQUESTED** [review]

File: core/src/main/scala/org/broadinstitute/dsde/rawls/model/SamModel.scala (hunk 2 above)
Comment (andy7i) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/model/SamModel.scala]: that makes sense! thanks!

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 4]
```diff
@@ -2105,6 +2105,9 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
...
               traceDBIOWithParent("setupGoogleProject", s1)(_ => DBIO.from(setupGoogleProject(billingProject, billingAccount, s1, workspaceId))).flatMap { case (googleProjectId, googleProjectNumber) =>
                 traceDBIOWithParent("saveNewWorkspace", s1)(s2 => saveNewWorkspace(workspaceId, workspaceRequest, bucketName, projectOwnerPolicyEmail, googleProjectId, Option(googleProjectNumber), Option(billingAccount), dataAccess, s2).flatMap { case (savedWorkspace, policyMap) =>
+                  // After the workspace has been created, create the google-project resource in Sam with the workspace as the resource parent
+                  traceDBIOWithParent("createResourceFull (google project)", s1)(_ => DBIO.from(samDAO.createResourceFull(SamResourceTypeNames.googleProject, googleProjectId.value, Map.empty, workspaceRequest.authorizationDomain.getOrElse(Set.empty).map(_.membersGroupName.value), userInfo, Option(SamFullyQualifiedResourceId(workspaceId, SamResourceTypeNames.workspace.value)))))
```
Comment (andy7i) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:2109]: this is the PR now 🙃

**Review by s-rubenstein — APPROVED** [review]
LGTM

**Review by dvoet — APPROVED** [review]

**Pull request merged**